server/yagna/init.sh

```diff
@@ -13,6 +13,10 @@ sleep 5
 /usr/bin/yagna payment fund --driver zksync
 /usr/bin/yagna payment init --sender --driver zksync
 
+# having key generation a last step allows api service to wait until yagna is ready
+/usr/bin/yagna app-key create requestor
```

Comment:
This command outputs the generated key. You could save this and skip the whole JSON parsing.

Reply:
Yes, actually this was [the 1st approach](https://github.com/golemfactory/yagna-service-poc/pull/5/commits/b454a6afeb83ea0d3a5d1288cfed7f5b20c8abdb#diff-654ade571040d742e696c7195dae43b8d7de4f328b4d31f9789db308da6240ac) however the file's content will be erased on subsequent call (starting yagna again on previously generated files)